go/runtime/host/sgx/ecdsa.go

```diff
@@ -105,6 +110,75 @@ func (ec *teeStateECDSA) Update(ctx context.Context, sp *sgxProvisioner, conn pr
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve TCB bundle: %w", err)
 	}
+	return tcbBundle, nil
+}
+
+func (ec *teeStateECDSA) Update(ctx context.Context, serviceStore *persistent.ServiceStore, sp *sgxProvisioner, conn protocol.Connection, report []byte, nonce string) ([]byte, error) {
+	rawQuote, err := sp.aesm.GetQuoteEx(ctx, ec.key, report)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get quote: %w", err)
+	}
+
+	var quote pcs.Quote
+	if err = quote.UnmarshalBinary(rawQuote); err != nil {
+		return nil, fmt.Errorf("failed to parse quote: %w", err)
+	}
+
+	var tcbBundle *pcs.TCBBundle
+	var cachedBundle *cachedTCBBundle
+	cacheDirty := false
+	bundleDownloaded := false
+
+	switch err := serviceStore.GetCBOR([]byte(tcbCacheKey), &cachedBundle); err {
+	case nil, persistent.ErrNotFound:
+		// Not cached yet. Not an error.
+	default:
+		// Can't get it... an error, but we can still try downloading it.
+		sp.logger.Warn("error checking common store for cached TCB bundle",
+			"err", err,
+		)
+	}
+
+	if cachedBundle == nil {
```

Comment:
It would be good if this bundle-cache logic was in a separate struct so it could be more easily unit-tested with various bundles/timestamps/etc.